pallets/identity/src/auth.rs

```diff
@@ -102,7 +101,8 @@ impl<T: Config> Module<T> {
             // If the sender is linked to an identity, ensure that it has relevant permissions
             pallet_permissions::Module::<T>::ensure_call_permissions(&sender)?.primary_did
         } else {
-            Context::current_identity_or::<Self>(&sender)?
+            // TODO: This should work for unlinked keys?
+            Self::get_identity(&sender).ok_or(Error::<T>::Unauthorized)?
```

Comment:
The old and new code would prevent unlinked keys from removing authorizations.  We do allow an unlinked key to call `remove_authorization` and have the creator of the auth pay for the transaction.  But this code would always throw an error.

Reply:
`Context::current_identity` would have returned the DID of that created the authorization in this case.  Which would cause `from_did` to be the same as `auth.authorized_by` below and `revoked` would be `true` instead of `false`.

Reply:
I fixed this by making `from_did` an Option.